Swift/C++ interop toolchain test feature.

adetaylor opened the pull request:
#### ee54eb308acdc9e465fa040679c2938bf6fdb05c
<pre>
Swift/C++ interop toolchain test feature.
<a href="https://bugs.webkit.org/show_bug.cgi?id=294029">https://bugs.webkit.org/show_bug.cgi?id=294029</a>
<a href="https://rdar.apple.com/153087725">rdar://153087725</a>

Reviewed by Elliott Williams.

This is a change which introduces a very simple Swift/C++ interop feature
into the core WebKit project. Specifically it intercepts a URI x-swift-demo://
which displays a Swift logo.

This is enabled by setting
  ENABLE_SWIFT_DEMO_URI_SCHEME = ENABLE_SWIFT_DEMO_URI_SCHEME
in LocalOverrides.xcconfig.

The purpose is to provide a simple test case to prove that Swift, and
specifically Swift/C++ interop, is working in a given toolchain. We want to
do this in advance of landing more meaningful Swift features in the core
of WebKit in future.

This feature will be removed in future once Swift is a well established
part of core WebKit.

This PR does not contain any cmake support: that will follow.

* Source/WebCore/platform/LegacySchemeRegistry.cpp:
(WebCore::builtinSecureSchemes):
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Modules/Internal/WebKitInternalCxx.h:
* Source/WebKit/Scripts/generate-plain-cpp-swift-interop-header.sh: Added.
* Source/WebKit/UIProcess/SwiftDemoLogo.swift: Added.
(Data.bytes):
(getSwiftLogoData):
* Source/WebKit/UIProcess/SwiftDemoLogoConfirmation.h: Copied from Source/WebKit/Modules/Internal/WebKitInternalCxx.h.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::decidePolicyForNavigationAction):
(WebKit::shouldShowSwiftDemoLogo):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
(WebKit::WebLoaderStrategy::scheduleLoad):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::platformCanHandleRequest):
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:

Canonical link: <a href="https://commits.webkit.org/299192@main">https://commits.webkit.org/299192@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ea17e23abbb209480be2de7ebbee273ea763eb0f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118209 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37887 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28526 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124368 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70249 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/dd34e46d-e22c-4194-8b4e-8adb5f005ba6) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120087 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38580 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46469 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89702 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59337 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/547501fd-d5fe-4d7b-ac0f-f709b46301e3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121162 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30731 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105988 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70198 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/14849776-2427-4cc9-a422-42976d1e31c3) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/117569 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29800 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24107 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68034 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/110320 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100157 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24281 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127442 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/116719 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45112 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34007 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98385 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45473 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102209 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98172 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24958 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43566 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21555 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41577 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44984 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50658 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/145419 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44444 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37411 "Found 1 new JSC binary failure: testapi, Found 19691 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/concat1.js.default, ChakraCore.yaml/ChakraCore/test/Array/push2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Basics/flags.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/blue_1086262.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/bug215238.shrua-2.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47789 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46133 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/Scripts/generate-plain-cpp-swift-interop-header.sh:
```diff
@@ -0,0 +1,26 @@
+#   to generate a header which can be safely included in pure C++ code.
+# This shell script does the second thing. We use the built-in XCode action
+# for the first.
+# Working around rdar://152836730
```
Not sure I fully understand the purpose of this script, but can it be removed entirely once this bug is fixed in our toolchains?

adetaylor replied:
Yes.

emw-apple commented on Source/WebKit/UIProcess/SwiftDemoLogo.swift:
```diff
@@ -0,0 +1,163 @@
+    if !WebKit_Internal_Cxx.WebKit.shouldShowSwiftDemoLogo() {
+        return []
+    }
+    // From "Download logo and guidelines" linked from
+    // https://developer.apple.com/swift/resources/
+    // Swift_logo_color.svg converted to PNG
```
I believe the Swift logo is Apache 2 licensed ([1], [2]), so you probably need to copy the attribution text from here: https://github.com/swiftlang/swift-org-website/blob/e31c5bf573d984ad98342e82033875071dab9473/LICENSE.txt#L178-L201

[1]: https://commons.wikimedia.org/wiki/File:Swift_logo.svg
[2]: https://github.com/swiftlang/swift-org-website/blob/main/LICENSE.md

adetaylor replied:
The specific download from which I got this file contains a "Swift Logo Guidelines.pdf" document stating the conditions under which it can be used. No license text is required. It does state that anywhere we mention Swift we should say "The Swift logo is a trademark of Apple Inc." I imagine that applies to lots of places in our codebase but I've added it here out of an abundance of caution.

adetaylor commented on Source/WebKit/Scripts/generate-plain-cpp-swift-interop-header.sh:
```diff
@@ -0,0 +1,26 @@
+# refers to only one other header. Once we expand beyond that, more sophistication
+# will be needed.
+
+/usr/bin/swiftc -typecheck -emit-clang-header-path "${SCRIPT_OUTPUT_FILE_0}" "${SCRIPT_INPUT_FILE_1}" -DENABLE_SWIFT_DEMO_URI_SCHEME \
```
Get correct swift path

adetaylor replied:
`xcrun swiftc`

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -19739,6 +19747,28 @@
+			inputPaths = (
+				"$(SRCROOT)/Scripts/generate-plain-cpp-swift-interop-header.sh",
+				"$(SRCROOT)/UIProcess/SwiftDemoLogo.swift",
+				"$(SRCROOT)/Modules/Internal/WebKitInternal.h",
```
Would it be possible to depend on the natively-generated swiftmodule, such that any time we rebuild WebKit_Internal.swiftmodule, we regenerate this header also?

Otherwise it seems like there are unavoidable cases where this could disrupt an incremental build, such as changes to headers transitively imported from `WebKitInternal.h`.

emw-apple replied:
Hm…alternatively you could investigate passing `-emit-dependencies` to swiftc and plumbing the file it generates back into Xcode by setting it a discovered dependency file for this script phase. That ought to be better at accurately capturing what the c++ header depends on.

I'm not entirely sure what the semantics for the path that `-emit-dependencies` çhooses. In native usage it looks like it's specified via the OutputFileMap.json.

Also, if you go down this route you'll have to emit an empty .d file when the feature is disabled. I usually do that using `echo "dependencies: " > /path/to/depfile.d`

adetaylor replied:
I spent quite a bit of time trying to make `-emit-dependencies` work (in [this tag](https://github.com/WebKit/WebKit/compare/main...adetaylor:WebKit:x-swift-demo-3-with-script-deps)). It generates a `.d` file but the contents of the `.d` file seem to be wrong/weird. I'd need to talk to the Swift team to diagnose why; I'll do so this week.

I am loathe to depend on the swift module since such a dependency may not be compatible with the order in which Xcode chooses to schedule operations.

Instead I looked through the generated .d file and have manually put in place the right stuff (corrected .h files and the modulemap)

I am strongly of the view that this is sufficient as we intend for this hacky script to be used only for the Swift demo logo stuff until upstream toolchain bugs are fixed. Can I convince you?

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -19739,6 +19747,28 @@
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "if [ \"${ENABLE_SWIFT_DEMO_URI_SCHEME}\" != \"ENABLE_SWIFT_DEMO_URI_SCHEME\" ]; then\nrm -f \"${SCRIPT_OUTPUT_FILE_0}\"\nelse\n\"${SCRIPT_INPUT_FILE_0}\"\nfi";
```
`rm -f` as a build step is quite uncommon for us; do you know if this is actually needed? I would think that it's fine to keep the header around, especially since the same build-time flag controls whether we are going to import it.

adetaylor replied:
Removed.

emw-apple commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -75,6 +76,10 @@ WK_SWIFT_WERROR_FLAGS_ = -Werror StrictMemorySafety;
 
+// Swift/C++ interop, by default, exports any symbols that are exposed from Swift to C++.
+// We don't want that.
+SWIFT_CXX_UNEXPORTED_SYMBOL_LDFLAGS = -Wl,-unexported_symbol,__ZN6WebKit16getSwiftLogoDataEv;
```
Do we have a bug tracking this? Seems like it should not be a requirement that the language exports any symbol that crosses the interop boundary.

I know about rdar://116086634, not sure if it's the same thing though.

adetaylor replied:
rdar://150187207 is the WebKit bug for this problem - I'll mention both in the comment.

emw-apple commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -24,8 +24,9 @@
-GCC_PREPROCESSOR_DEFINITIONS = $(ENABLE_RELOCATABLE_WEBPUSHD) $(inherited) $(GCC_PREPROCESSOR_DEFINITIONS_$(ENABLE_WEBGPU_SWIFT));
+GCC_PREPROCESSOR_DEFINITIONS = $(ENABLE_RELOCATABLE_WEBPUSHD) $(inherited) $(GCC_PREPROCESSOR_DEFINITIONS_$(ENABLE_WEBGPU_SWIFT)) $(GCC_PREPROCESSOR_DEFINITIONS_$(ENABLE_SWIFT_DEMO_URI_SCHEME));
 GCC_PREPROCESSOR_DEFINITIONS_ENABLE_WEBGPU_SWIFT = ENABLE_WEBGPU_SWIFT;
+GCC_PREPROCESSOR_DEFINITIONS_ENABLE_SWIFT_DEMO_URI_SCHEME = ENABLE_SWIFT_DEMO_URI_SCHEME;
```
nit: I think the ENABLE_WEBGPU_SWIFT one is needlessly wordy. Since build-webkit feature flags are defined as their own feature name, we can do this: 
```suggestion
GCC_PREPROCESSOR_DEFINITIONS = $(ENABLE_RELOCATABLE_WEBPUSHD) $(inherited) $(ENABLE_WEBGPU_SWIFT) $(ENABLE_SWIFT_DEMO_URI_SCHEME);
```

adetaylor replied:
Worked well, thanks!

emw-apple commented on Source/WebKit/Modules/Internal/WebKitInternalCxx.h:
```diff
@@ -25,4 +25,5 @@
 
 // Add project-level C++ header files here to be able to access them from within Swift sources.
 
+#import <UIProcess/SwiftDemoLogoConfirmation.h>
```
Interesting that this is a bracket-style import, since it's not a public/private API header. Does it not work as a quote import, and if so why?

adetaylor replied:
Fixed.

emw-apple commented on Source/WebKit/Scripts/generate-plain-cpp-swift-interop-header.sh:
```diff
@@ -0,0 +1,28 @@
+# This is mostly OK because right now the only header file included from Swift
+# refers to only one other header.
+
+xcrun swiftc -typecheck -emit-clang-header-path "${SCRIPT_OUTPUT_FILE_0}" "${SCRIPT_INPUT_FILE_1}" -DENABLE_SWIFT_DEMO_URI_SCHEME \
```
Could we pass in all of OTHER_SWIFT_FLAGS here?
```suggestion
xcrun swiftc -typecheck -emit-clang-header-path "${SCRIPT_OUTPUT_FILE_0}" "${SCRIPT_INPUT_FILE_1}" ${OTHER_SWIFT_FLAGS} \
```

Among other things, that will give access to the full set of Platform.h defines.

adetaylor replied:
This didn't work for some reason - it didn't contain `-DENABLE_SWIFT_DEMO_URI_SCHEME` but it did seem to contain other flags which confused the swiftc command-line parsing.

How much do you care about this? IMO we _want_ this script to be primitive and limited so that people aren't tempted to use it, before we replace it with the built-in Xcode functionality when the known bugs are fixed. I don't want people using this beyond the Swift demo logo use-case, so this limitation seems to me to be actively a good thing.

emw-apple replied:
Strange, I wonder if the `generate-platform-args` script is not parsing GCC_PREPROCESSOR_DEFINITIONS like we'd expect.

But yeah, I agree that this isn't so important that it needs to be addressed. We can treat this script as a hacky workaround for ENABLE_SWIFT_DEMO_URI_SCHEME, and we can extend it further if changing Swift doesn't pan out and we're stuck with this approach.

adetaylor replied:
I looked into it. `WebKitBuild/Debug/DerivedSources/WebKit/platform-enabled-swift-args.arm64e.resp` does contain `-DENABLE_SWIFT_DEMO_URI_SCHEME`. However `OTHER_SWIFT_FLAGS` actually asks this script to `@` include `WebKitBuild/Debug/DerivedSources/WebKit/platform-enabled-swift-args.undefined_arch.resp` which doesn't exist at all.

I thought that adding `runOncePerArchitecture = 1` to the relevant part of the `pbxproj` might solve this, but it doesn't seem to. In any case the built-in Xcode functionality to emit such a header does not seem to be architecture-specific so I'm not keen on making an arch-specific header file.

Also, the specific error I see is:

```
error: unexpected input file: @/path/WebKitBuild/Debug/DerivedSources/WebKit/platform-enabled-swift-args.undefined_arch.resp
```

which suggests that `swiftc` in this mode doesn't even support `resp` files.

If you know a magic way to fix all this let me know and I'm happy to do in a follow-up PR. Marking as resolved meanwhile.

emw-apple commented on Source/WebKit/UIProcess/SwiftDemoLogo.swift:
```diff
@@ -0,0 +1,164 @@
+/// Return data for a PNG of a Swift logo.
+@available(iOS 15.0, macOS 15.0, tvOS 15.0, watchOS 9.0, *)
+@_expose(Cxx)
+public func getSwiftLogoData() -> [UInt8] {
```
Out of curiosity, does this function end up in WebKit's public swiftmodule when this feature is turned on? That seems like a problem for real adoption.

adetaylor replied:
`strings ./WebKitBuild/Debug/WebKit.framework/Versions/A/Modules/WebKit.swiftmodule/arm64e-apple-macos.swiftmodule | grep getSwiftLogoData` says yes.

I agree this is no good for real adoption. I've filed bugs related to this in the past but this is a nice crisp case so I might use it as an excuse to file another one - I'll look into the related bugs tomorrow.

Are you agreed that it does not need to block this PR?

emw-apple replied:
Presumably it's in the textual .swiftinterface too, I think that's what actually matters. Yes, this shouldn't block this PR.

adetaylor replied:
OK. Good news. By adding `@_spi(Internal)` I can move it from `.swiftinterface` to `.private.swiftinterface` which hopefully means we're good here. Pinged you on chat to make sure, but I'm going to mark this conversation as resolved meanwhile.

rr-codes commented on Source/WebKit/UIProcess/SwiftDemoLogo.swift:
```diff
@@ -0,0 +1,162 @@
+
+import Foundation
+internal import WebKit_Internal_Cxx
+import struct Swift.String
```
I established a pattern in `Foundation+Extras.swift` which typealiases String to Swift.String

```
typealias String = Swift.String
```

so this line is not needed

adetaylor replied:
Ah cool. Fixed.

rr-codes commented on Source/WebKit/UIProcess/SwiftDemoLogo.swift:
```diff
@@ -0,0 +1,162 @@
+/// Return data for a PNG of a Swift logo.
+@_expose(Cxx)
+@_spi(Private)
+public func getSwiftLogoData() -> [UInt8] {
```
Hm, this will become actual WebKit SPI by doing this, which I don't think we want?

adetaylor replied:
We definitely don't, but it's slightly better than a public API. I added a FIXME linking to the radar which explains why those are currently the two options.

rr-codes commented on Source/WebKit/UIProcess/SwiftDemoLogoConfirmation.h:
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2014-2025 Apple Inc. All rights reserved.
```
is 2014 right?

adetaylor replied:
Fixed

rr-codes commented on Source/WebKit/UIProcess/SwiftDemoLogo.swift:
```diff
@@ -0,0 +1,162 @@
+@_expose(Cxx)
+@_spi(Private)
+public func getSwiftLogoData() -> [UInt8] {
+    if !WebKit_Internal_Cxx.WebKit.shouldShowSwiftDemoLogo() {
```
```suggestion
    guard WebKit.shouldShowSwiftDemoLogo() else {
        return []
    }
```

adetaylor replied:
Done, thanks! Good idea

rr-codes commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -16794,6 +16817,18 @@ Ref<AboutSchemeHandler> WebPageProxy::protectedAboutSchemeHandler()
+#else
+    // This shouldn't even be called if ENABLE_SWIFT_DEMO_URI_SCHEME
+    // isn't enabled, but we'll return false for good measure.
+    return false;
```
should this just be `RELEASE_ASSERT_NOT_REACHED()` in that case?

adetaylor replied:
Done

adetaylor committed:
Swift/C++ interop toolchain test feature.
https://bugs.webkit.org/show_bug.cgi?id=294029
rdar://153087725

Reviewed by Elliott Williams.

This is a change which introduces a very simple Swift/C++ interop feature
into the core WebKit project. Specifically it intercepts a URI x-swift-demo://
which displays a Swift logo.

This is enabled by setting
  ENABLE_SWIFT_DEMO_URI_SCHEME = ENABLE_SWIFT_DEMO_URI_SCHEME
in LocalOverrides.xcconfig.

The purpose is to provide a simple test case to prove that Swift, and
specifically Swift/C++ interop, is working in a given toolchain. We want to
do this in advance of landing more meaningful Swift features in the core
of WebKit in future.

This feature will be removed in future once Swift is a well established
part of core WebKit.

This PR does not contain any cmake support: that will follow.

* Source/WebCore/platform/LegacySchemeRegistry.cpp:
(WebCore::builtinSecureSchemes):
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Modules/Internal/WebKitInternalCxx.h:
* Source/WebKit/Scripts/generate-plain-cpp-swift-interop-header.sh: Added.
* Source/WebKit/UIProcess/SwiftDemoLogo.swift: Added.
(Data.bytes):
(getSwiftLogoData):
* Source/WebKit/UIProcess/SwiftDemoLogoConfirmation.h: Copied from Source/WebKit/Modules/Internal/WebKitInternalCxx.h.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::decidePolicyForNavigationAction):
(WebKit::shouldShowSwiftDemoLogo):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
(WebKit::WebLoaderStrategy::scheduleLoad):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::platformCanHandleRequest):
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:

Canonical link: https://commits.webkit.org/299192@main
